tools/DevicesApiTester/Commands/Gpio/GpioButtonEvent.cs

```diff
@@ -0,0 +1,119 @@
+
+        [Option('l', "led-pin", HelpText = "The GPIO pin the LED is connected to (if any). The number is based on the --scheme argument.", Required = false)]
+        public int? LedPin { get; set; }
+
+        [Option('p', "pressed-value", HelpText = "The value of the GPIO pin when the button is pressed: { Rising | Falling }", Required = false, Default = PinEventTypes.Rising)]
+        public PinEventTypes PressedValue { get; set; }
+
+        [Option("on-value", HelpText = "The value that turns the LED on: { High | Low }", Required = false, Default = PinValue.High)]
+        public PinValue OnValue { get; set; }
+
+        /// <summary>Executes the command asynchronously.</summary>
+        /// <returns>The command's exit code.</returns>
+        /// <remarks>
+        ///     NOTE: This test app uses the base class's <see cref="CreateGpioController"/> method to create a device.<br/>
+        ///     Real-world usage would simply create an instance of <see cref="GpioController"/>:
+        ///     <code>using (var gpio = new GpioController())</code>
+        /// </remarks>
+        public Task<int> ExecuteAsync()
+        {
+            if (LedPin != null)
+            {
+                Console.WriteLine($"Driver={Driver}, Scheme={Scheme}, ButtonPin={ButtonPin}, LedPin={LedPin}, PressedValue={PressedValue}, OnValue={OnValue}");
+            }
+            else
+            {
+                Console.WriteLine($"Driver={Driver}, Scheme={Scheme}, ButtonPin={ButtonPin}, PressedValue={PressedValue}, OnValue={OnValue}");
+            }
+
+            using (GpioController gpio = CreateGpioController())
+            {
+                using (var cancelEvent = new ManualResetEvent(false))
+                {
+                    Console.WriteLine($"Listening for button presses on GPIO {Enum.GetName(typeof(PinNumberingScheme), Scheme)} pin {ButtonPin} . . .");
+
+                    // This example runs until Ctrl+C (or Ctrl+Break) is pressed, so register a local function handler.
+                    Console.CancelKeyPress += Console_CancelKeyPress;
+                    gpio.OpenPin(ButtonPin);
+
+                    // Set the mode based on if input pull-up resistors are supported.
+                    PinMode inputMode = gpio.IsPinModeSupported(ButtonPin, PinMode.InputPullUp) ? PinMode.InputPullUp : PinMode.Input;
+                    gpio.SetPinMode(ButtonPin, inputMode);
+
+                    // Open the GPIO pin connected to the LED if one was specified.
+                    if (LedPin != null)
+                    {
+                        gpio.OpenPin(LedPin.Value, PinMode.Output);
+                        gpio.Write(LedPin.Value, OffValue);
+                    }
+
+                    // Declare a local function to handle the pin value changed events.
+                    int count = 0;
+                    void valueChangeHandler(object sender, PinValueChangedEventArgs pinValueChangedEventArgs)
```

Comment:
I know you didn't add this local function, but could you please push it to the end of this method j ust like the other local function? Sometimes it's hard to read these, so usually having them after the return statement of the function makes it a bit more readable.

Reply:
This was mainly a file rename.  I reordered the display of properties for better flow, but nothing logic -wise changed.